core/src/banking_stage.rs

```diff
@@ -1166,16 +1267,31 @@ impl BankingStage {
         );
         process_tx_time.stop();
         let unprocessed_tx_count = unprocessed_tx_indexes.len();
+        inc_new_counter_info!(
+            "banking_stage-unprocessed_transactions",
+            unprocessed_tx_count
+        );
 
         // applying cost of processed transactions to shared cost_tracker
-        transactions.iter().enumerate().for_each(|(index, tx)| {
-            if !unprocessed_tx_indexes.iter().any(|&i| i == index) {
-                let tx_cost = cost_model.read().unwrap().calculate_cost(tx.transaction());
-                let mut guard = cost_tracker.lock().unwrap();
-                let _result = guard.try_add(tx_cost);
-                drop(guard);
-            }
-        });
+        let mut cost_tracking_time = Measure::start("cost_tracking_time");
+        let mut tx_cost = TransactionCost::new_with_capacity(MAX_WRITABLE_ACCOUNTS);
+        {
+            //let cost_model_readonly = cost_model.read().unwrap();
+            //let mut cost_tracker_mutable = cost_tracker.write().unwrap();
+            transactions.iter().enumerate().for_each(|(index, tx)| {
+                if !unprocessed_tx_indexes.iter().any(|&i| i == index) {
```

Comment:
How big we expect `transcations` and `unprocessed_tx_indexes` be here?
Wouldn't this cause an `O(n^2)` performance cost?